sycl/plugins/level_zero/usm_allocator.cpp

```diff
@@ -280,11 +261,17 @@ class Bucket {
   SystemMemory &getMemHandle();
   USMAllocContext::USMAllocImpl &getUsmAllocCtx() { return OwnAllocCtx; }
 
-  bool BucketHasCapacity();
+  bool CanPool();
 
 private:
   void onFreeChunk(Slab &);
+
+  // Get a slab to be used for chunked allocations.
+  // These slabs are used for allocations <= ChunkCutOff and not pooled.
```

Comment:
> not pooled.

Why is it that we don't pull some allocations?

Reply:
For allocations below a slabMinSize there is already amortization across multiple allocations because they share a slab. There is diminishing returns over pooling these small allocations. For instance if 8-byte allocations were pooled, only 1 out of ~8000 allocations will lead to a allocation from the pool vs from the system.
And, the implementation will change quite a bit to support chunked and pooled allocations. I didn't think it was worth it.

Reply:
Can you out some of this explanation as a source code comment?